e2e/specs/age-based-vitals.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { expect } from '@playwright/test';
+import { test } from '../core';
+import { BiometricsAndVitalsPage } from '../pages';
+import { type Patient, deletePatient, endVisit, generateRandomPatient, startVisit } from '../commands';
+import { type Visit } from '@openmrs/esm-framework';
+
+const createdPatients: Patient[] = [];
+const createdVisits: Visit[] = [];
+
+function calculateBirthdate(age: { years?: number; months?: number }): string {
+  const date = new Date();
+  if (age.years) date.setFullYear(date.getFullYear() - age.years);
+  if (age.months) date.setMonth(date.getMonth() - age.months);
+  return date.toISOString().split('T')[0];
```

Comment:
Probably better not to use `toISOString()`. This will not always return the same date as calculated as it's dependent on the user's timezone (`toISOString()` first moves the date to UTC).